cypress/integration/Homework/calculator.spec.js

```diff
@@ -0,0 +1,203 @@
+                performCalculatorOperation(numberArray.firstSymbol, numberArray.secondSymbol, 'Divide')
+                checkIfErrorFieldContainsMessage(errorMessage)
+            })
+        })
+
+        describe(formDescribeSentence("3", buildVersion, "Calculator 'Clear' button functionality testing with numbers"), () => {
+
+            beforeSectionForPageAndIntegers(numberArray.integersOnly, false, buildVersion)
+
+            beforeEach("Selects the 'Integers only' marker based on the integersOnly value", () => {
+
+                clearBothInputFields()
+
+                if (numberArray.integersOnly) {
+                    calculator.integerSelection().click()
+                }
+            })
+
+            it(formItSentence("3", "1", "Add", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Add')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "2", "Subtract", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Subtract')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "3", "Multiply", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Multiply')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "4", "Divide", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field or gets the '" + divideByZeroError + "' error"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Divide')
+                checkDivisionAnswerOrError(numberArray, false, buildVersion, divideByZeroError, true)
+            })
+
+            it(formItSentence("3", "5", "Concatenate", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Concatenate')
+                clearAndCheckAnswerField(true)
+            })
+        })
+    })
+})
+
+function visitPageSelectBuildIntegersOnly(integersOnly, integerValidation, buildVersion) {
+    calculator.visitInitialPage()
+    calculator.buildDropDownList().select(buildVersion)
+
+    if (integersOnly && integerValidation) {
+        calculator.integerSelection().click()
+    }
+}
+
+function beforeSectionForPageAndIntegers(integersOnly, integerValidation, buildVersion) {
```

Comment:
No need for this function. Use simple before above the tests. Easier to add more items later on.